File: sbe/binance/spot_3_2/generated-python/spot_sbe/GroupSizeEncoding.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple
from io import BytesIO

class GroupSizeEncoding:
    """Composite type."""

    def __init__(self):
        self.block_length = 0
        self.num_in_group = 0

    def encode(self, buffer: BytesIO) -> None:
        """Encode the composite to the buffer."""
        buffer.write(struct.pack('<H', self.block_length))
        buffer.write(struct.pack('<I', self.num_in_group))

    def decode(self, buffer: BytesIO) -> None:
        """Decode the composite from the buffer."""
        self.block_length = struct.unpack('<H', buffer.read(2))[0]
        self.num_in_group = struct.unpack('<I', buffer.read(4))[0]

    @staticmethod
    def encoded_length() -> int:
        return 6

File: sbe/binance/stream_1_0/generated-php/spot_stream/GroupSizeEncoding.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class GroupSizeEncoding
{
    public int|float $blockLength = 0;
    public int|float $numInGroup = 0;

    public function encode(): string
    {
        $buffer = '';

        $buffer .= pack('v', $this->blockLength);
        $buffer .= pack('V', $this->numInGroup);

        return $buffer;
    }

    public function decode(string $data, int &$offset = 0): void
    {
        $this->blockLength = unpack('v', substr($data, $offset, 2))[1];
        $offset += 2;
        $this->numInGroup = unpack('V', substr($data, $offset, 4))[1];
        $offset += 4;
    }

    public static function encodedLength(): int
    {
        return 6;
    }
}
